contracts/minter/src/contract.rs

```diff
@@ -443,6 +453,39 @@ fn _execute_mint(
         .add_messages(msgs))
 }
 
+fn random_mintable_token_id(deps: Deps, env: &Env, sender: Addr) -> Result<u32, ContractError> {
+    // not cryptographically secure random
+    // TODO add random beacon input to sha256 step
+    let transaction_index: u32 = if let Some(transaction) = &env.transaction {
+        transaction.index
+    } else {
+        return Err(ContractError::NoEnvTransactionIndex {});
+    };
+    let height: u64 = env.block.height;
+    let sha256 = Sha256::digest(format!("{}{}{}", sender, transaction_index, height).into_bytes());
+    let randomness: Vec<u8> = sha256.to_vec();
+    let randomness: [u8; 16] = randomness[0..16].try_into().unwrap(); // Cut first 16 bytes from 32 byte value
+
+    // A PRNG that is not cryptographically secure.
+    // See https://docs.rs/rand/0.8.5/rand/rngs/struct.SmallRng.html
+    // where this is used for 32 bit systems.
+    // We don't use the SmallRng in order to get the same implementation
+    // in unit tests (64 bit dev machines) and the real contract (32 bit Wasm)
+    let mut rng = Xoshiro128PlusPlus::from_seed(randomness);
+    let mintable_tokens_result: StdResult<Vec<u32>> = MINTABLE_TOKEN_IDS
+        .keys(deps.storage, None, None, Order::Ascending)
+        .collect();
+    let mut mintable_tokens = mintable_tokens_result.unwrap();
+    if mintable_tokens.is_empty() {
+        return Err(ContractError::SoldOut {});
+    }
```

Comment:
can we move this at the top and return early if it's sold out to avoid extra compute work.

Reply:
https://github.com/public-awesome/stargaze-contracts/pull/208/commits/a112243a7b8e8b0cc4b17665b657064570cc075d
https://github.com/public-awesome/stargaze-contracts/pull/208/commits/c02dcedc8b644ea108f35669ae6afa2a0123ca08